app/src/main/java/unipd/se18/ocrcamera/PreProcessing.java

```diff
@@ -0,0 +1,394 @@
+        Mat modifiedMat = new Mat();
+        switch (isBright(bright)) {
+            case IMAGE_TOO_BRIGHT:
+                Log.d(TAG, "Case==IMAGE_TOO_BRIGHT");
+                //Darkens the colour's brightness until it's in an optimal value
+                for(double changeBrightness = 0; changeBrightness != maxBrightness;
+                    changeBrightness -= step){
+                   /*This variable is used to select the type of matrix we want to abtain in the
+                      the convertTo method. If it's negative the type doesn't change*/
+                    int matrixType = -1;
+
+                    /*This variable is used to change the contrast of the matrix, but we want
+                      only modify the brightness so we put the value 1 because the method use
+                      this formula from the documentation:
+                      m(x,y) = saturate _ cast<rType>(alpha(*this)(x,y) + beta)
+                      We called beta as changeBrightness*/
+                    int alpha = 1;
+
+                    /*Modify the values of all pixels with an alpha and beta value following
+                      the formula above.*/
+                    bright.convertTo(modifiedMat, matrixType, alpha, changeBrightness);
+
+                    //Verify if the image is good enough
+                    if(isBright(modifiedMat) == BrightnessValue.IMAGE_IS_OK){
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case IMAGE_TOO_DARK:
+                Log.d(TAG, "Case==IMAGE_TOO_DARK");
+                //The variables are explained in the case above
+                //Lightens the colour's brightness until it's in an optimal value
+                for(double changeBrightness = 0; changeBrightness != maxBrightness;
+                    changeBrightness += step){
+
+                    //Converts an array to another data type with optional scaling. The variables's
+                    // values are explained in case 1
+                    int matrixType = -1;
+                    int alpha = 1;
+                    bright.convertTo(modifiedMat, matrixType, alpha, changeBrightness);
+
+                    //Verify if the image is good enough
+                    if(isBright(modifiedMat) == BrightnessValue.IMAGE_IS_OK){
+                        //If the conversion failed it returns the original image
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
```

Comment:
as in the other method(the cropping one) here you have no way of knowing if it failed or not from outside